src/screens/ManageTag/ManageTag.tsx

```diff
@@ -45,78 +55,97 @@ function ManageTag(): JSX.Element {
   });
   const { t: tCommon } = useTranslation('common');
 
-  const [unassignTagModalIsOpen, setUnassignTagModalIsOpen] = useState(false);
-
+  const [unassignUserTagModalIsOpen, setUnassignUserTagModalIsOpen] =
+    useState(false);
   const [addPeopleToTagModalIsOpen, setAddPeopleToTagModalIsOpen] =
     useState(false);
-  const [assignToTagsModalIsOpen, setAssignToTagsModalIsOpen] = useState(false);
-
-  const [editTagModalIsOpen, setEditTagModalIsOpen] = useState(false);
-  const [removeTagModalIsOpen, setRemoveTagModalIsOpen] = useState(false);
+  const [tagActionsModalIsOpen, setTagActionsModalIsOpen] = useState(false);
+  const [editUserTagModalIsOpen, setEditUserTagModalIsOpen] = useState(false);
+  const [removeUserTagModalIsOpen, setRemoveUserTagModalIsOpen] =
+    useState(false);
 
   const { orgId, tagId: currentTagId } = useParams();
   const navigate = useNavigate();
-  const [after, setAfter] = useState<string | null | undefined>(null);
-  const [before, setBefore] = useState<string | null | undefined>(null);
-  const [first, setFirst] = useState<number | null>(5);
-  const [last, setLast] = useState<number | null>(null);
-
   const [unassignUserId, setUnassignUserId] = useState(null);
 
   // a state to specify whether we're assigning to tags or removing from tags
   const [tagActionType, setTagActionType] =
     useState<TagActionType>('assignToTags');
 
   const toggleRemoveUserTagModal = (): void => {
-    setRemoveTagModalIsOpen(!removeTagModalIsOpen);
+    setRemoveUserTagModalIsOpen(!removeUserTagModalIsOpen);
   };
 
   const showAddPeopleToTagModal = (): void => {
     setAddPeopleToTagModalIsOpen(true);
   };
-
   const hideAddPeopleToTagModal = (): void => {
     setAddPeopleToTagModalIsOpen(false);
   };
 
-  const showAssignToTagsModal = (): void => {
-    setAssignToTagsModalIsOpen(true);
+  const showTagActionsModal = (): void => {
+    setTagActionsModalIsOpen(true);
   };
-
-  const hideAssignToTagsModal = (): void => {
-    setAssignToTagsModalIsOpen(false);
+  const hideTagActionsModal = (): void => {
+    setTagActionsModalIsOpen(false);
   };
 
-  const showEditTagModal = (): void => {
-    setEditTagModalIsOpen(true);
+  const showEditUserTagModal = (): void => {
+    setEditUserTagModalIsOpen(true);
   };
-
-  const hideEditTagModal = (): void => {
-    setEditTagModalIsOpen(false);
+  const hideEditUserTagModal = (): void => {
+    setEditUserTagModalIsOpen(false);
```

Comment:
A good idea, but the toggle and show require additional state modifications as well, will look at how it can be optimised later. For now, let's keep it as it is.